### ISSUE TITLE
Won't run because it sticks to trying to redeem oil

### ISSUE THREAD

**Issue by AkumaNoTsubasa** [LmeSzinc/AzurLaneAutoScript#4193]
### Before you asking

- [X] I have searched the existing issues
- [X] I spend at least 5 minutes for thinking and preparing
- [X] I checked Frequently Asked Questions (FAQ) in Wiki page
- [X] I am using the latest version of Alas

### Describe the bug

Tried to use this tool but all it does is going to the commission screen and clicking on the "Complete"-Button. Sadly I have more oil than I can accept so of course this does not work. I tried setting the scheduler for commissions to in two days but it sticks to doing nothing else than trying to redeem the reward that contains oil. After 12 tries it logs the game out, logs back in and repeats itself. Sadly I could not find a "if can't redeem rewards, go to next function" or anything similar.

### To Reproduce

Clicking "Start" on the scheduler

### Expected behavior

Expectation is, that if oil is full either spend current oil or just ignore the function that wants to give you oil

### Relevant log output

```Text
Logs:
INFO     02:02:22.526 │ Click ( 487,  286) @ REWARD_1_WHITE                        
INFO     02:02:25.751 │ Click ( 494,  286) @ REWARD_1_WHITE                        
INFO     02:02:29.055 │ Click ( 478,  290) @ REWARD_1_WHITE                        
INFO     02:02:32.357 │ Click ( 470,  293) @ REWARD_1_WHITE                        
INFO     02:02:35.665 │ Click ( 457,  292) @ REWARD_1_WHITE                        
INFO     02:02:38.975 │ Click ( 460,  292) @ REWARD_1_WHITE                        
INFO     02:02:42.281 │ Click ( 444,  287) @ REWARD_1_WHITE                        
INFO     02:02:45.590 │ Click ( 427,  293) @ REWARD_1_WHITE                        
INFO     02:02:48.893 │ Click ( 437,  290) @ REWARD_1_WHITE                        
INFO     02:02:52.209 │ Click ( 427,  294) @ REWARD_1_WHITE                        
INFO     02:02:55.512 │ Click ( 476,  291) @ REWARD_1_WHITE      
INFO     02:02:58.819 │ Function calls:                                            
                   <string>   L1 <module>                                          
                   spawn.py L105 spawn_main()                                      
                   spawn.py L118 _main()                                           
                 process.py L297 _bootstrap()                                      
                 process.py  L99 run()                                             
         process_manager.py L152 run_process()                                     
                    alas.py L537 loop()                                            
                    alas.py  L69 run()                                             
                    alas.py L190 commission()                                      
              commission.py L581 run()                                             
              commission.py L540 commission_receive()                              
                    base.py L189 appear_then_click()                               
                 control.py  L37 click()                                           
                  device.py L250 handle_control_check()                            
                  device.py L286 click_record_check()                              
                  device.py  L43 show_function_call()                              
WARNING  02:02:58.822 │ Too many click for a button: REWARD_1_WHITE                
WARNING  02:02:58.822 │ History click: ['REWARD_1_WHITE', 'REWARD_1_WHITE',        
         'REWARD_1_WHITE', 'REWARD_1_WHITE', 'REWARD_1_WHITE', 'REWARD_1_WHITE',   
         'REWARD_1_WHITE', 'REWARD_1_WHITE', 'REWARD_1_WHITE', 'REWARD_1_WHITE',   
         'REWARD_1_WHITE', 'REWARD_1_WHITE']                                       
ERROR    02:02:58.823 │ GameTooManyClickError: Too many click for a button:        
         REWARD_1_WHITE                                                            
WARNING  02:02:58.824 │ Saving error: ./log/error/1726272178824                    
WARNING  02:02:59.013 │ Game stuck, com.YoStarEN.AzurLane will be restarted in 10  
         seconds
```

### Screenshots

![image](https://github.com/user-attachments/assets/a35d7dea-cb88-4da4-abfa-8fe1b463d821)

**Comment by tsubami8** [comment]
maybe cause by the new battle theme,i meet the error too and when i change back to the default battle theme It's running normally

**Comment by AkumaNoTsubasa** [comment]
Can you maybe elaborate what you mean exactly?

**Comment by LmeSzinc** [comment]
buy curry in dorm to spend oil

**Comment by AkumaNoTsubasa** [comment]
> buy curry in dorm to spend oil

But that would only be a temporary solution and you hit the spot of "too much oil" quite often. Can't we add a simple failsafe that "if can't redeem, ignore for now" or just add a check "If oil >= capacity don't redeem oil stuff"?

**Comment by LmeSzinc** [comment]
Enable some tasks and run ALAS everyday, you shouldn't have that much oil overflowed, just need a temp fix to pull the trigger.

**Comment by AkumaNoTsubasa** [comment]
I am trying but getting under 4500 oil is really hard. I managed to get below 20k and after restarting the game today I am now back to 25k and since all that time I was absolutely unable to run ALAS because the redemption of the oil redeem has a so high priority that it alas won't do anything unless I am below 4500 oil so it can redeem.

**Comment by AkumaNoTsubasa** [comment]
![image](https://github.com/user-attachments/assets/4fccfd9b-259c-4657-9760-20e9a16027cd)
I tried with my second account I started a few weeks ago where I have less oil and it has another full lockup. It doesn't register I am not advanced enough to do that commission and it sticks to that commission and tries it 10 times, then logs me out, logs back in and does the same again.
So I can't use the bot on both of my accounts because it locks itself into loops.
